control-plane/gateways/deployment_init_container.go

```diff
@@ -30,15 +32,15 @@ type initContainerCommandData struct {
 	LogJSON  bool
 }
 
-// containerInit returns the init container spec for connect-init that polls for the service and the connect proxy service to be registered
+// initContainer returns the init container spec for connect-init that polls for the service and the connect proxy service to be registered
 // so that it can save the proxy service id to the shared volume and boostrap Envoy with the proxy-id.
-func initContainer(config GatewayConfig, name, namespace string) (corev1.Container, error) {
+func (b *meshGatewayBuilder) initContainer() (corev1.Container, error) {
 	data := initContainerCommandData{
-		AuthMethod:         config.AuthMethod,
-		LogLevel:           config.LogLevel,
-		LogJSON:            config.LogJSON,
-		ServiceName:        name,
-		ServiceAccountName: name,
+		AuthMethod:         b.config.AuthMethod,
+		LogLevel:           b.config.LogLevel,
+		LogJSON:            b.config.LogJSON,
+		ServiceName:        b.gateway.Name,
+		ServiceAccountName: b.gateway.Name,
```

Comment:
Like to use this everywhere since we have it -- in theory, changing the naming logic would then only have to happen in one place.

```suggestion
		ServiceAccountName: b.serviceAccountName(),
```